xapian-letor/ranker/ranker.cc

```diff
@@ -22,13 +22,280 @@
 
 #include <config.h>
 
+#include "xapian-letor/featurelist.h"
+#include "xapian-letor/featurevector.h"
+#include "xapian-letor/letor_error.h"
 #include "xapian-letor/ranker.h"
```

Comment:
Our convention is to include the header corresponding to the source file first in that source file (well, second really - `<config.h>` comes first).

The reason for this is that it means that every header (unless it has no corresponding source file) gets included once without anything else having been included first, so we know that it must be self-contained, rather than happening to work because some other header is always included first.

So this ought to be:

```
#include <config.h>

#include "xapian-letor/ranker.h"
 
#include "xapian-letor/featurelist.h"
#include "xapian-letor/featurevector.h"
#include "xapian-letor/letor_error.h"
#include "xapian-letor/scorer.h"
```